lib/fuelsdk/soap.rb

```diff
@@ -99,9 +99,23 @@ def wsdl
       @wsdl ||= 'https://webservice.exacttarget.com/etframework.wsdl'
     end
 
+    def check_soap_client_for_refresh
+      if auth_token_expiration.nil? || Time.new + 480 > self.auth_token_expiration
+        self.refresh!
+        new_savon_client
+      end
+    end
+
     def soap_client
-      self.refresh
+      if auth_token_expiration.nil? || Time.new + 300 > self.auth_token_expiration
```

Comment:
can this part of the fi statement be replaced by a call to check_soap_cleint_for_refresh (if you want the added times to be different you can make it a param with a default value of 480)